src/scripting/GameHooks.h

```diff
@@ -7,22 +7,29 @@ struct GameMainThread
 
     static GameMainThread& Get();
 
-    void AddTask(std::function<void()> aFunction);
+    void AddBaseInitializationTask(const std::function<bool()>& aFunction);
+    void AddInitializationTask(const std::function<bool()>& aFunction);
+    void AddRunningTask(const std::function<bool()>& aFunction);
+    void AddShutdownTask(const std::function<bool()>& aFunction);
+    void AddGenericTask(const std::function<bool()>& aFunction);
 
     ~GameMainThread();
 
 private:
     GameMainThread();
-    
+
     void Hook();
     void Unhook() const;
 
-    using TMainThreadLoop = bool(void*, void*);
+    using TMainThreadStateTick = bool(RED4ext::IGameState*, RED4ext::CGameApplication*);
```

Comment:
A struct should be more appropriate here

Reply:
dont know what you mean by this exactly, changed the rest 🤔

Reply:
don't use std::pair, first and second are not expressive, it's hard to understand what they are

Reply:
oh, this one
kk, i got confused cause it shows this comment on another line 😄

Reply:
cleaned it up, think this looks rather nice now 👍

Reply:
actually, wait a bit - want to do one more thing

Reply:
k think this is really done now 😁